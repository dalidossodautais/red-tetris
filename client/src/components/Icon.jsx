import React from "react";
import Tooltip from "@material-ui/core/Tooltip";
import Build from "@material-ui/icons/Build";
import ExpandLess from "@material-ui/icons/ExpandLess";
import FastForward from "@material-ui/icons/FastForward";
import FastRewind from "@material-ui/icons/FastRewind";
import FormatListNumbered from "@material-ui/icons/FormatListNumbered";
import Lock from "@material-ui/icons/Lock";
import People from "@material-ui/icons/People";
import Settings from "@material-ui/icons/Settings";
import SportsEsports from "@material-ui/icons/SportsEsports";
import Star from "@material-ui/icons/Star";
import StarBorder from "@material-ui/icons/StarBorder";
import VideogameAsset from "@material-ui/icons/VideogameAsset";
import { makeStyles } from "@material-ui/core/styles";
import Visibility from "@material-ui/icons/Visibility";

function useClasses(marginRight, onClick) {
  return makeStyles(() => ({
    icon: {
      cursor: onClick ? "default" : "pointer",
      marginRight,
    },
  }))();
}

export default function Icon({ marginRight, onClick, tooltip, type }) {
  const classes = useClasses(marginRight, onClick);
  const Icon =
    {
      admin: Star,
      gamepad: VideogameAsset,
      keep: ExpandLess,
      nonAdmin: StarBorder,
      people: People,
      player: SportsEsports,
      private: Lock,
      ranking: FormatListNumbered,
      settings: Settings,
      speedDown: FastRewind,
      speedUp: FastForward,
      viewer: Visibility,
    }[type] || Build;
  return (
    <Tooltip title={tooltip || ""}>
      <Icon className={classes.icon} onClick={onClick} />
    </Tooltip>
  );
}
