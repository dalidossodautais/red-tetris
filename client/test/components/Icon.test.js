import React from "react";
import renderer from "react-test-renderer";
import Icon from "../../src/components/Icon";

describe("Icon", () => {
  test("Standard", () => {
    const component = renderer.create(<Icon />);
    // rendering
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Standard with onClick", () => {
    const component = renderer.create(<Icon onClick={() => true} />);
    // rendering
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
