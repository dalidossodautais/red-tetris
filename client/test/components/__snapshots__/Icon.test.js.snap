// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Standard 1`] = `
<svg
  aria-describedby={null}
  aria-hidden={true}
  className="MuiSvgIcon-root makeStyles-icon-1"
  focusable="false"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title=""
  viewBox="0 0 24 24"
>
  <path
    d="M22.7 19l-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
  />
</svg>
`;

exports[`Icon Standard 2`] = `
<svg
  aria-describedby={null}
  aria-hidden={true}
  className="MuiSvgIcon-root makeStyles-icon-1"
  focusable="false"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title=""
  viewBox="0 0 24 24"
>
  <path
    d="M22.7 19l-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
  />
</svg>
`;

exports[`Icon Standard with onClick 1`] = `
<svg
  aria-describedby={null}
  aria-hidden={true}
  className="MuiSvgIcon-root makeStyles-icon-2"
  focusable="false"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title=""
  viewBox="0 0 24 24"
>
  <path
    d="M22.7 19l-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
  />
</svg>
`;

exports[`Icon Standard with onClick 2`] = `
<svg
  aria-describedby={null}
  aria-hidden={true}
  className="MuiSvgIcon-root makeStyles-icon-2"
  focusable="false"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title=""
  viewBox="0 0 24 24"
>
  <path
    d="M22.7 19l-9.1-9.1c.9-2.3.4-5-1.5-6.9-2-2-5-2.4-7.4-1.3L9 6 6 9 1.6 4.7C.4 7.1.9 10.1 2.9 12.1c1.9 1.9 4.6 2.4 6.9 1.5l9.1 9.1c.4.4 1 .4 1.4 0l2.3-2.3c.5-.4.5-1.1.1-1.4z"
  />
</svg>
`;
